server/api/subscribe_handler.py

```diff
@@ -4,40 +4,101 @@
 from config import STRIPE_API_KEY
 from utils.auth.middleware import check_token
 
-create_subscription_handler = Blueprint('create_subscription_handler', __name__)
+subscription_handler = Blueprint('create_subscription_handler', __name__)
 stripe.api_key = STRIPE_API_KEY
 
 
 def create_customer(payment_method_id, email):
     try:
         customer = stripe.Customer.create(
             payment_method=payment_method_id,
-            email=email
+            email=email,
+            invoice_settings={
+                'default_payment_method': payment_method_id,
+            },
         )
-        stripe.Customer.create_source(
-            customer['id'],
-            source="tok_mastercard",
-        )
-        # TODO: record customer id in db
+        # stripe.Customer.create_source(
+        #     customer['id'],
+        #     source="tok_mastercard",
+        # )
         return customer['id']
-    except:
-        return Exception
+    except Exception as e:
+        return None
 
 
-@create_subscription_handler.route('/create-subscription', methods=['POST'])
+@subscription_handler.route('/create-subscription', methods=['POST'])
 @check_token
 def create_subscription():
     try:
         data = json.loads(request.data)
-        # TODO: check if user already had customer id in db
-        customer_id = create_customer(data['payment_method_id'], data['email'])
-        # else:
-        stripe.Subscription.create(
+
+        # Check if customer exists
+        customer_list = stripe.Customer.list(email = data.get("email")).get("data")
+
+        customer_id = None
+
+        if len(customer_list) > 0:
+            customer_id = customer_list[0].get("id")
+        else: # If not exist create customer
+            customer_id = create_customer(data.get('payment_method_id'), data.get('email'))
+        
+        if not customer_id:
+            raise Exception("Error creating stripe customer")
```

Comment:
I guess the `customer_id` in the top should be `customer_id=False` so that the ` if not customer_id:` will work

Reply:
Good catch!

Reply:
Just realized change the if condition into if customer_id is None would be better